ngraph/src/ngraph/frontend/onnx_import/op/dequantize_linear.cpp

```diff
@@ -33,54 +33,195 @@ namespace ngraph
+                                         scale_dim,
+                                         "' must match the input shape dimension '",
+                                         x_dim_at_axis,
+                                         " pointed to by the axis attribute: ",
+                                         axis);
+                        }
                     }
 
-                    if (x->get_element_type() != zero_point->get_element_type())
+                    void validate_zero_point(const std::shared_ptr<ngraph::Node> zero_point,
+                                             const std::shared_ptr<ngraph::Node> x,
+                                             const int64_t axis)
                     {
-                        zero_point = std::make_shared<default_opset::Convert>(
-                            zero_point, x->get_element_type());
+                        const auto& zero_point_shape = zero_point->get_output_partial_shape(0);
+                        NGRAPH_CHECK(zero_point_shape.rank().get_length() == 0 ||
+                                         zero_point_shape.rank().get_length() == 1,
+                                     "Zero point needs to be a scalar or a vector.");
+
+                        if (zero_point_shape.rank().get_length() == 1)
+                        {
+                            const auto& zero_point_dim = zero_point_shape[0];
+                            const auto& x_shape = x->get_output_partial_shape(0);
+                            const auto& x_dim_at_axis = x_shape[axis];
+
+                            NGRAPH_CHECK(zero_point_dim.same_scheme(x_dim_at_axis),
+                                         "The number of zero point elements '",
+                                         zero_point_dim,
+                                         "' must match the input shape dimension '",
+                                         x_dim_at_axis,
+                                         " pointed to by the axis attribute: ",
+                                         axis);
+                        }
                     }
 
-                    return {std::make_shared<ngraph::opset0::Dequantize>(
-                        x, x_scale, zero_point, x_scale->get_element_type(), axes)};
+                    std::shared_ptr<ngraph::Node>
+                        reshape_input(const std::shared_ptr<ngraph::Node> input,
+                                      const int64_t axis,
+                                      const PartialShape& x_shape)
+                    {
+                        std::vector<int64_t> target_dims;
+
+                        for (size_t i = 0; i < axis; ++i)
+                        {
+                            target_dims.push_back(1);
+                        }
+
+                        // copy dimension at axis from input X
+                        target_dims.push_back(x_shape[axis].get_length());
```

Comment:
You're checking below only that the rank of `x_shape` is static and here you're accessing specific dimension value - this may cause errors in case of dynamic dimension.